### DISCUSSION TITLE
React-Native-dapps-v2explorer-Errors // SOLVED

### DISCUSSION THREAD

**Discussion by Mustafa6900** [WalletConnect/walletconnect-monorepo#2365] [Developer Support] [answered]
Hello, I want to use the connect wallet button in my project. I tried the example you shared (https://github.com/WalletConnect/react-native-examples/tree/main/dapps/v2Explorer). I got an error like this and couldn't find a solution.

 ERROR  Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.   
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes

**Comment by chendinh** [comment]
Can you tell more step by step, what did you do ?

**Comment by Mustafa6900** [comment]
Hi,

Step 1: I added expo to the project with "npm i expo".
Step2: I changed the script in the package json as follows.
"scripts": {
   "android:build": "cd android && ./gradlew assembleRelease",
    "lint": "eslint .",
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "test": "jest"
  },
Step3: I changed the .env and .env.example files as follows. (I got the
project id from walletconnect.cloud.)
ENV_PROJECT_ID=2c43a08ce0737aa905523a2b1c845629
ENV_RELAY_URL=wss://relay.walletconnect.com


4:npm start
chendinh ***@***.***>, 15 May 2023 Pzt, 06:14 tarihinde şunu
yazdı:

**Comment by finessevanes** [comment]
I've escalated this to our RN team!

**Comment by Mustafa6900** [comment]
Thanks, I've been trying to reach people who are authorized or
knowledgeable for 1 month. I will be very happy if you solve my problems.

19 May 2023 Cum 07:33 tarihinde vanes ***@***.***> şunu yazdı:

**Comment by ignaciosantise** [comment]
Hey @Mustafa6900 👋  What are you trying to accomplish by adding `expo` module? The `v2explorer` example is bare react-native, and there's no way to turn it into an expo project.

If you want to create a new project with expo, follow the steps in `Expo Go Quickstart` tab in [react-native docs](https://reactnative.dev/docs/environment-setup?guide=quickstart), 

And then you can check our docs to use web3modal with expo: https://docs.walletconnect.com/2.0/reactnative/web3modal/Installation

**Comment by Mustafa6900** [comment]
I added the expo so that I can open the emulator. I couldn't open any other
kind of emulator.

19 May 2023 Cum 15:39 tarihinde ignaciosantise ***@***.***>
şunu yazdı:

**Comment by Mustafa6900** [comment]
When I load the library you suggested with Expo, I get a different error.
Can you explain the steps in more detail?

19 May 2023 Cum 15:39 tarihinde ignaciosantise ***@***.***>
şunu yazdı:

**Comment by ignaciosantise** [comment, marked as answer]
The example works without adding more libraries. Try cloning the repo again and follow this [guide](https://github.com/WalletConnect/react-native-examples/tree/main/dapps/v2Explorer#installation) to install & run the example.

**Comment by Mustafa6900** [comment]
: Task failed with an exception.
-----------
* Where:
Build file
'C:\Users\Musti\Desktop\react-native-examples\dapps\v2Explorer\android\app\build.gradle'
line: 2

* What went wrong:
A problem occurred evaluating project ':app'.
> Could not find implementation class 'com.facebook.react.ReactPlugin' for
plugin 'com.facebook.react' specified in
jar:file:/C:/Users/Musti/.gradle/caches/jars-9/53fbca66afc5503918bd8298c4572a02/react-native-gradle-plugin.jar!/META-INF/gradle-plugins/com.facebook.react.properties.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring project ':app'.
> compileSdkVersion is not specified. Please add it to build.gradle

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BU�LD FAILED in 47s

Error: Command failed: gradlew.bat app:installDebug
-PreactNativeDevServerPort=8081

FAILURE: Build completed with 2 failures.

ignaciosantise ***@***.***>, 19 May 2023 Cum, 15:59 tarihinde
şunu yazdı:

**Comment by Mustafa6900** [comment]
Bro thanks <3

For those wondering how I fixed this error:

...\react-native-examples\dapps\v2Explorer\android

I opened admin cmd inside the folder.

step 1) gradlew --stop

step 2) gradlew clean

step 3) gradlew.bat app:installDebug -PreactNativeDevServerPort=8081

I ran your commands and it fixed the problem.

Mustafa Demirhan ***@***.***>, 19 May 2023 Cum, 17:54
tarihinde şunu yazdı:

**Comment by ignaciosantise** [comment]
That's great @Mustafa6900! Closing the discussion as the problem was solved 🚀
